json_serialization/reader.nim

```diff
@@ -187,7 +187,7 @@ proc readJsonNodeField(r: var JsonReader, field: var JsonNode)
   field = r.parseJsonNode()
 
 proc parseJsonNode(r: var JsonReader): JsonNode =
-  const maxIntValue = BiggestInt.high.uint64 + 1
+  const maxIntValue: uint64 = BiggestInt.high.uint64 + 1
```

Comment:
`BiggestInt` means signed int?
A recurrent question: are we sure this doesn't overflow? :)

Reply:
Ooh good one! `BiggestInt` is an alias for `int64` as per
https://github.com/status-im/Nim/blob/3c9b68dc157804885b14a1984efc25e8b7cc861d/lib/system/basic_types.nim#L36-L39

So yes, it means the biggest type of signed int :)
Do you think it's better to use int64 instead?

Reply:
> Do you think it's better to use int64 instead?

Yes good point! I think is better to explicitly use `int64` to make sure it won't fail in the future if the `BiggestInt` definition changes to `int128` for instance, if that type ever exists in the future. 

Aside from that, will that work on 32 bit architecture? I'm thinking whether the `unit32` and `int32` types will suffice for us.

wdyt? @etan-status @gabrielmer

Reply:
Please disregard this comment:
> Aside from that, will that work on 32 bit architecture? I'm thinking whether the unit32 and int32 types will suffice for us.

`int64` can work well in 32-bit architecture too.

Reply:
I have kind of mixed feelings.

On one hand, I like the naming of `BiggestInt` because of explaining that we intentionally use the biggest `int` possible and the hypothetical case of an int128 can be far fetched (although not negligible I guess).

On the other, yes, if at some point there gets to exist an `int128` it we'll have a bug. It would remind us to change the code to support the new type promptly, but it would fail at the beginning and that's never good.

@etan-status what's your take?

Reply:
can we use `BiggestUint` (or equivalent) instead of `uint64`? that would also solve the problem.

If it doesn't work, can put a `static: doAssert sizeof(uint64) >= sizeof(BiggestInt)` or similar, to have a compile-time check that the assumption holds. If it ever changes, there would be a compile-time error and you would get alerted to it.

Reply:
I like the solution! I switched the uses of `uint64` and `int64` to `BiggestUint` and `BiggestInt` respectively. Everything worked for me locally, hope it passes the CI.

Please let me know if this looks ok and if so then we should be good to go :))

Reply:
> Please let me know if this looks ok and if so then we should be good to go :))

LGMT! Thank you guys! :clap: